src/safeposix/syscalls/net_calls.rs

```diff
@@ -1955,6 +1993,7 @@ impl Cage {
                 continue;
             }
 
+            // get the FileDescriptor Object from fd
             let checkedfd = self.get_filedescriptor(fd).unwrap();
             let unlocked_fd = checkedfd.read();
             if let Some(filedesc_enum) = &*unlocked_fd {
```

Comment:
Can you explain a bit about what this condition is doing with some comments, currently not able to understand its use case.